### PR TITLE
Add `swift experimental-destination install` subcommand

### PR DESCRIPTION
**Pull request by MaxDesiatov**
Depends on https://github.com/apple/swift-package-manager/pull/6127.

### Motivation:

Cross-compilation destinations pitch includes `install` subcommand for installing artifact bundles.

### Modifications:

Added new `InstallDestination` subcommand that takes a single argument for a bundle to install from.

### Result:

`swift experimental-destination install` is available for installing destination bundles.

### CODE REVIEW

File: Sources/CrossCompilationDestinationsTool/InstallDestination.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            }
+
+            let fileName = bundleURL.lastPathComponent
+
+            try fileSystem.writeFileContents(destinationsDirectory.appending(component: fileName), data: body)
+        } else if
+            let cwd = fileSystem.currentWorkingDirectory,
+            let bundlePath = try? AbsolutePath(validating: bundlePathOrURL, relativeTo: cwd)
+        {
+            try fileSystem.move(from: bundlePath, to: destinationsDirectory)
```
Comment (tomerd) [inline Sources/CrossCompilationDestinationsTool/InstallDestination.swift:76]: move or copy?

File: Sources/CrossCompilationDestinationsTool/InstallDestination.swift [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        // FIXME: generalize path calculation and creation with `ListDestinations` subcommand
+        if !fileSystem.exists(destinationsDirectory) {
+            destinationsDirectory = try fileSystem.getOrCreateSwiftPMCrossCompilationDestinationsDirectory()
+        }
+
+        let observabilitySystem = ObservabilitySystem(
+            SwiftToolObservabilityHandler(outputStream: stdoutStream, logLevel: .warning)
+        )
```
Comment (tomerd) [inline Sources/CrossCompilationDestinationsTool/InstallDestination.swift:57]: if this is a repeated pattern we can potential define a  SwiftToolObservabilitySystem that takes optional output stream and log level but defaults to stdout and warn

**Review by neonichu — APPROVED** [review]

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
https://github.com/apple/sourcekit-lsp/pull/711

@swift-ci test Windows

**Comment by MaxDesiatov** [comment]
https://github.com/apple/sourcekit-lsp/pull/711

@swift-ci test Windows

**Comment by MaxDesiatov** [comment]
@swift-ci smoke test

**Comment by MaxDesiatov** [comment]
https://github.com/apple/sourcekit-lsp/pull/711

@swift-ci test Windows

**Pull request merged**